src/util/swrite.cc

```diff
@@ -53,3 +59,42 @@ int swrite( int fd, const char *str, ssize_t len )
 
   return 0;
 }
+
+
+int swrite_timeout( int fd, uint64_t timeout_ms, const char *str, ssize_t len )
```

Comment:
Well, the sole reason for it being there, is the blocking socket write (without timeout) that already was just next to it. While not very elegant, the reasoning there was that should agent connection in the client end block, it should after a reasonably short timeout just fail so that the process that tries to use it via forwarding, can continue its authentication process using other means.
If someone wishes to write this in some other way, please do, but that's more or less workable.